cedar-policy-formatter/src/pprint/utils.rs

```diff
@@ -24,27 +24,27 @@ pub fn add_brackets<'a>(d: RcDoc<'a>, leftp: RcDoc<'a>, rightp: RcDoc<'a>) -> Rc
     leftp.append(d.nest(1)).append(rightp)
 }
 
+/// Convert a leading comment to an `RcDoc`, adding leading and trailing newlines.
 pub fn get_leading_comment_doc_from_str<'a>(leading_comment: &str) -> RcDoc<'a> {
     if leading_comment.is_empty() {
         RcDoc::nil()
     } else {
-        let cs: RcDoc<'_> = RcDoc::intersperse(
-            leading_comment
-                .trim()
-                .split('\n')
-                .map(|c| RcDoc::text(c.to_owned())),
-            RcDoc::hardline(),
-        );
-        RcDoc::hardline().append(cs).append(RcDoc::hardline())
+        RcDoc::hardline()
+            .append(create_multiline_doc(leading_comment))
+            .append(RcDoc::hardline())
     }
 }
 
-pub fn get_trailing_comment_doc_from_str<'a>(trailing_comment: &str) -> RcDoc<'a> {
+/// Convert a trailing comment to an `RcDoc`, adding a trailing newline.
+pub fn get_trailing_comment_doc_from_str<'a>(
+    trailing_comment: &str,
+    next_doc: RcDoc<'a>,
+) -> RcDoc<'a> {
     if trailing_comment.is_empty() {
-        RcDoc::nil()
+        next_doc
     } else {
         RcDoc::space()
-            .append(RcDoc::text(trailing_comment.trim().to_owned()))
+            .append(create_multiline_doc(trailing_comment))
```

Comment:
I'm surprised that this doesn't causes changes in any test cases. Also, I'm not sure what impact, if any, this change actually should have